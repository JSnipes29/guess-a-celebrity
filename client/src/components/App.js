/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import {
  useNavigate, BrowserRouter as Router, Routes, Route,
} from 'react-router-dom';
import Box from '@material-ui/core/Box';
import '../assets/App.css';
import '../assets/styles.css';
import LoginForm from './LoginForm';
import GamePage from './GamePage';
import Score from './ScorePage';

function Game({ user, scores, setUser }) {
  const navigate = useNavigate();
  const routeChange = (path, state) => {
    navigate(path, state);
  };
  return (
    <div>
      <Routes>
        <Route exact path="/" element={<GamePage scores={scores} user={user} routeChange={routeChange} />} />
        <Route path="/score" element={<Score user={user} setUser={setUser} routeChange={routeChange} />} />
      </Routes>
    </div>
  );
}

function App() {
  // user
  const [user, setUser] = useState({ name: '' });
  // the scores to show while playing
  const [maxScores, setScores] = useState({
    userMax: 0,
    globalMax: 0,
    globalName: '',
  });
  return (
    <div className="App">
      {(user.name === '') ? (
        <div>
          <h1 style={{ marginBottom: '2.5em' }}>Guess That Celebrity</h1>
          <Box style={{ top: '50%' }} color="white" bgcolor="palevioletred" p={1}>
            <div>
              Enter your name:
            </div>
          </Box>
          <LoginForm setScores={setScores} setUsers={setUser} />
        </div>
      ) : (
        <Router>
          <Game scores={maxScores} user={user} setUser={setUser} />
        </Router>
      ) }
    </div>
  );
}

export default App;
