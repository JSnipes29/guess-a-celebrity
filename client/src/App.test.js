/**
 * @jest-environment jsdom
 */

/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import App from './components/App';

function GamePage() {
  const [user, setUser] = useState({ name: '' });
  const [maxScores] = useState({
    userMax: 0,
    globalMax: 0,
    globalName: '',
  });
  return (<App.Game scores={maxScores} user={user} setUser={setUser} />);
}
test('renders title', () => {
  render(<App />);
  const linkElement = screen.getByText(/Guess That Celebrity/i);
  expect(linkElement).toBeInTheDocument();
});

test('login button', () => {
  render(<App />);
  const linkElement = screen.getByText(/Login/i);
  expect(linkElement).toBeInTheDocument();
});

test('snapshot test', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
