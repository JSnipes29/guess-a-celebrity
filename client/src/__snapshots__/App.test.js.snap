// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
>
  <div>
    <h1
      style={
        Object {
          "marginBottom": "2.5em",
        }
      }
    >
      Guess That Celebrity
    </h1>
    <div
      className="MuiBox-root MuiBox-root-3"
      style={
        Object {
          "top": "50%",
        }
      }
    >
      <div>
        Enter your name:
      </div>
    </div>
    <div>
      <div>
        <div>
          Name:
        </div>
        <input
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <input
        className="button"
        onClick={[Function]}
        type="submit"
        value="login"
      />
    </div>
  </div>
</div>
`;
